src/Workspaces/CSharp/Portable/Extensions/MemberDeclarationSyntaxExtensions.cs

```diff
@@ -284,19 +284,13 @@ public static ArrowExpressionClauseSyntax GetExpressionBody(this MemberDeclarati
             switch (memberDeclaration?.Kind())
             {
                 case SyntaxKind.PropertyDeclaration:
-                    return ((PropertyDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.MethodDeclaration:
-                    return ((MethodDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.IndexerDeclaration:
-                    return ((IndexerDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.OperatorDeclaration:
-                    return ((OperatorDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.ConversionOperatorDeclaration:
-                    return ((ConversionOperatorDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.ConstructorDeclaration:
-                    return ((ConstructorDeclarationSyntax)memberDeclaration).ExpressionBody;
                 case SyntaxKind.DestructorDeclaration:
-                    return ((DestructorDeclarationSyntax)memberDeclaration).ExpressionBody;
+                    return ((BaseMethodDeclarationSyntax)memberDeclaration).ExpressionBody;
```

Comment:
this won't work (afaict).  For example, IndexerDeclaration and PropertyDeclaration do not derive from BaseMethodDeclarationSyntax.

Reply:
Ah, good point. I checked most of the nodes, and tested it locally, and it passed, though, I guess there isn't coverage for those. I've reverted, don't think it adds that much value, in this case to refactor, especially if it requires adding a lot of "synthetic" coverage.